server/src/main/java/org/apache/druid/server/QueryScheduler.java

```diff
@@ -108,6 +140,7 @@ public QueryScheduler(
       this.totalCapacity = serverConfig.getNumThreads();
     }
     this.laneRegistry = BulkheadRegistry.of(getLaneConfigs(limitTotal));
+    this.emitter = new ServiceEmitter("test", "localhost", new NoopEmitter());
```

Comment:
So this ctor is for the test case only? I think we'd better not instantiate such object here but pass an object from the test cases.

Reply:
agree, have removed the ctor and the object is now directly initiated from the test classes